### PR TITLE
Fixes and Changes 

### PR DESCRIPTION
**Pull request by TorchedSammy**
Here is yet another PR, I added the master branch of my fork this time and reversed the more personal changes.
- Icon is retrieved by both extension and scope (.h is correctly marked as C)
- `time_per_file` option (#34)
- `big_icon` is default 
- Add icons for Kotlin, Git, Swift, Assembly, XML, remove D 
- Idle icon which is also used on startup
- Gets the last and correct extension (before something like `index.d.ts` reported `d` instead of `ts`, which is incorrect)
- Actually use `start_state` (#37)

There's a bit more but those are the main things

### CODE REVIEW

File: drp.py [hunk 1]
```diff
@@ -22,63 +22,104 @@
...
+    'rb': 'ruby',
+    'rs': 'rust',
+    'sh': 'shell',
+    'swift': 'swift',
+    't': 'perl',
+    'toml': 'toml',
+    'ts': 'typescript',
+    'txt': 'text',
+    'vue': 'vue',
+    'xml': 'xml'
```
Comment (Snazzah) [inline drp.py:80]: Should add `.svg` (Or let it have it's own icon)

**Review by Snazzah — CHANGES_REQUESTED** [review]

**Comment by Snazzah** [comment]
Any specific reason D lang support was dropped?

**Comment by TorchedSammy** [comment]
ah i forgot to change back the readme :| no reason that Dlang was dropped besides that I didn't make an icon for it, ill add it back and you can use the icon you have currently. ill work on an icon after merge

**Comment by Snazzah** [comment]
> ah i forgot to change back the readme :| no reason that Dlang was dropped besides that I didn't make an icon for it, ill add it back and you can use the icon you have currently. ill work on an icon after merge

No need, I've already made a one that should work with the style.

**Comment by TorchedSammy** [comment]
very nice. made all the changes you requested

**Pull request merged**